packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts

```diff
@@ -262,47 +402,67 @@ export class AdapterDayjs implements MuiPickersAdapter<Dayjs, string> {
   };
 
   public isSameYear = (value: Dayjs, comparing: Dayjs) => {
-    return value.isSame(comparing, 'year');
+    return this.isSame(value, comparing, 'YYYY');
   };
 
   public isSameMonth = (value: Dayjs, comparing: Dayjs) => {
-    return value.isSame(comparing, 'month');
+    return this.isSame(value, comparing, 'YYYY-MM');
   };
 
   public isSameDay = (value: Dayjs, comparing: Dayjs) => {
-    return value.isSame(comparing, 'day');
+    return this.isSame(value, comparing, 'YYYY-MM-DD');
   };
 
   public isSameHour = (value: Dayjs, comparing: Dayjs) => {
     return value.isSame(comparing, 'hour');
   };
 
   public isAfter = (value: Dayjs, comparing: Dayjs) => {
-    return value.isAfter(comparing);
+    return value > comparing;
   };
 
   public isAfterYear = (value: Dayjs, comparing: Dayjs) => {
-    return value.isAfter(comparing, 'year');
+    if (!this.hasUTCPlugin()) {
+      return value.isAfter(comparing, 'year');
+    }
+
+    // @ts-ignore
+    return !this.isSameYear(value, comparing) && value.utc() > comparing.utc();
   };
 
   public isAfterDay = (value: Dayjs, comparing: Dayjs) => {
-    return value.isAfter(comparing, 'day');
+    if (!this.hasUTCPlugin()) {
+      return value.isAfter(comparing, 'day');
+    }
+
+    // @ts-ignore
+    return !this.isSameDay(value, comparing) && value.utc() > comparing.utc();
   };
 
   public isBefore = (value: Dayjs, comparing: Dayjs) => {
-    return value.isBefore(comparing);
+    return value < comparing;
   };
 
   public isBeforeYear = (value: Dayjs, comparing: Dayjs) => {
-    return value.isBefore(comparing, 'year');
+    if (!this.hasUTCPlugin()) {
+      return value.isBefore(comparing, 'year');
+    }
+
+    // @ts-ignore
+    return !this.isSameYear(value, comparing) && value.utc() < comparing.utc();
   };
 
   public isBeforeDay = (value: Dayjs, comparing: Dayjs) => {
-    return value.isBefore(comparing, 'day');
+    if (!this.hasUTCPlugin()) {
+      return value.isBefore(comparing, 'day');
+    }
+
+    // @ts-ignore
+    return !this.isSameDay(value, comparing) && value.utc() < comparing.utc();
   };
 
   public isWithinRange = (value: Dayjs, [start, end]: [Dayjs, Dayjs]) => {
-    return value.isBetween(start, end, null, '[]');
+    return value >= start && value <= end;
```

Comment:
We don't need the power of `isBefore` and `isAfter` because we are not passing any unit.
The main advantage of this change is that `isBetween` / `isAfter` are very slow on dayjs with the `timezone` plugin, so I'm trying to use them as little as possible